src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp.Client/Pages/Auth.razor

```diff
@@ -0,0 +1,18 @@
+﻿@page "/auth"
+
+@using Microsoft.AspNetCore.Authorization
+
+@attribute [Authorize]
+@*#if (UseServer && !InteractiveAtRoot)
+@attribute [RenderModeInteractiveAuto]
+##elseif (!InteractiveAtRoot)
+@attribute [RenderModeInteractiveWebAssembly]
+##endif*@
+
+<PageTitle>Auth</PageTitle>
+
+<h1>You are authenticated</h1>
+
+<AuthorizeView>
+    Hello @context.User.Identity?.Name!
```

Comment:
How can `context.User.Identity` be null if they are authenticated? Could this expression safely be changed to the following?

```suggestion
    Hello @context.User.Identity!.Name!
```

It's awkward that the two `!` symbols have completely different meanings here...

Reply:
That's part of the reason for using the `?` even though it should never be null on this page. Would you rather move the `!` from the end?

Reply:
It's not a big deal. I wrote this at an early stage of the code review when I was still picking up on minor details.